core/src/bundle/usBundleRegistry.cpp

```diff
@@ -49,7 +49,23 @@ BundleRegistry::~BundleRegistry(void)
 
 std::shared_ptr<Bundle> BundleRegistry::Register(BundleInfo* info)
 {
-  std::shared_ptr<Bundle> bundle(GetBundle(info->name));
+  std::vector<std::string> embeddedBundles; // statically-linked bundles in the binary
+  if (info->name.empty() || info->name == US_CORE_FRAMEWORK_NAME)
+  {
+    embeddedBundles = GetBundleNamesFromLibrary(info->location);
+  }
+  if(!embeddedBundles.empty())
+  {
+    if(info->name.empty())
+      info->name = embeddedBundles.at(0);
```

Comment:
braces should be used even for one line conditional statements to avoid subtle errors.